### PR TITLE
Prefer discrete GPU for PowerPreference::Default

### PR DESCRIPTION
**Pull request by aloucks**

### CODE REVIEW

**Comment by kvark** [comment]
I think it's more complex than that. We should check if we are on AC power, or on battery, and make this decision based off that. Would you want to explore this direction?

**Comment by aloucks** [comment]
The motivation behind this was that I have both discrete and integrated GPU enabled on my desktop and all of WGPU examples end up running on the integrated. WGPU punishes me for enabling the integrated GPU :) This differs from pretty much every other game/app I've run that always prefers the discrete GPU as the default option. 

It looks like the [battery](https://docs.rs/battery/0.7.5/battery/) crate could be used to check power status, although I'm not sure this approach is the most user friendly. It might be confusing if things perform differently based on whether or not the app was *started* while plugged in. Applications could detect changes in battery status and re-query adapters, but I doubt that would be common.

If you still want to go that route, I can amend the PR to check battery status. The battery crate  supports the following platforms:

* Linux 2.6.39+
* MacOS 10.10+
* Windows 7+
* FreeBSD
* DragonFlyBSD

**Comment by kvark** [comment]
Yes, please, let's do it. No need to react to changes, this is just to resolve what Default means at a particular point.

File: wgpu-core/Cargo.toml [hunk 1]
```diff
@@ -46,3 +46,6 @@ gfx-backend-vulkan = { version = "0.4", features = ["x11"] }
 gfx-backend-dx12 = { version = "0.4.1" }
 gfx-backend-dx11 = { version = "0.4" }
 gfx-backend-vulkan = { version = "0.4" }
+
+[target.'cfg(any(target_os = "linux", target_os = "macos", target_os = "windows", target_os = "dragonfly", target_os = "freebsd"))'.dependencies]
+battery = "0.7"
```
Comment (kvark) [inline wgpu-core/Cargo.toml]: nit: newline missing?

File: wgpu-core/Cargo.toml [hunk 2]
```diff
@@ -46,3 +46,6 @@ gfx-backend-vulkan = { version = "0.4", features = ["x11"] }
 gfx-backend-dx12 = { version = "0.4.1" }
 gfx-backend-dx11 = { version = "0.4" }
 gfx-backend-vulkan = { version = "0.4" }
+
+[target.'cfg(any(target_os = "linux", target_os = "macos", target_os = "windows", target_os = "dragonfly", target_os = "freebsd"))'.dependencies]
```
Comment (kvark) [inline wgpu-core/Cargo.toml]: does battery simply not build on other platforms?

Comment (aloucks) [reply wgpu-core/Cargo.toml]: It does not:

https://github.com/svartalf/rust-battery/blob/bed111f51f02d172f26ed00fa6a73ea51e084a60/battery/src/platform/mod.rs#L27-L28

Comment (kvark) [reply wgpu-core/Cargo.toml]: They could have just mocked it on their side, returning an Error...

**Pull request merged**